Dockerfile

```diff
@@ -236,12 +277,16 @@ RUN julia --project=${RAY_JL_PROJECT} -e 'using Pkg; Pkg.precompile(strict=true,
 
 FROM ray-base as ray-jl
 
-COPY --from=build-ray-jl --link $HOME/anaconda3 $HOME/anaconda3
-COPY --chown=${UID} --from=build-ray-jl --link ${JULIA_USER_DEPOT} ${JULIA_USER_DEPOT}
-RUN sudo chown ${UID} ${HOME} && \
-    ln -s ${JULIA_USER_DEPOT} ~/.julia
+# Equivalent of `python -m site --user-base`
+ARG PYTHON_USER_BASE=${HOME}/.local
+
+COPY --from=build-ray-jl --link ${PYTHON_USER_BASE} ${PYTHON_USER_BASE}
+ENV PATH="${PYTHON_USER_BASE}/bin:${PATH}"
+RUN ray --version
 
 ARG RAY_JL_PROJECT=${JULIA_USER_DEPOT}/dev/Ray
+COPY --from=build-ray-jl --chown=${UID} --link ${JULIA_USER_DEPOT} ${JULIA_USER_DEPOT}
+RUN julia --project=${RAY_JL_PROJECT} -e 'using Ray'
 
 # Set up default project and working dir so that users need only pass in the requisite script input args
 ENV JULIA_PROJECT=${RAY_JL_PROJECT}
```

Comment:
why not set the ENV and WORKDIR before calling `RUN julia --project=${RAY_JL_PROECT}`?

Reply:
We could. As this Dockerfile uses multiple projects I think it's clearer to not use `JULIA_PROJECT` within this Dockerfile. Setting these at the end are just conveniences for interactive Docker image use.